pkg/tools/list/graph.go

```diff
@@ -52,84 +53,64 @@ func NewDirectedAcyclicGraph() *DirectedAcyclicGraph {
 	return &DirectedAcyclicGraph{Graph: *NewGraph()}
 }
 
-type Identifier interface {
-	Id() string
-}
-
-type IdentifierSet struct {
-	ids map[string]Identifier
-}
-
-func (s *IdentifierSet) Add(id Identifier) {
-	s.ids[id.Id()] = id
-}
-
-func (s *IdentifierSet) Remove(id string) {
-	delete(s.ids, id)
-}
-
-func (s *IdentifierSet) Contains(id string) bool {
-	_, ok := s.ids[id]
-	return ok
-}
-
-func (s *IdentifierSet) Size() int {
-	return len(s.ids)
-}
-
 // Vertex defines the node with in a graph with a unique id for index
 type Vertex interface {
-	Identifier
+	ds.Identifier
```

Comment:
Vertex and Edge use interface seems complication. keep the data struct simple is better, use struct is the first choice. the Vertex value field may be Object struct or interface.

Reply:
fixed, ptal

Reply:
removed the general graph/vertex/edge interface and instead use a simple `importGraph` struct without introducing any complicated data structure. ptal